### PR TITLE
Hotfix current week Number Calculation

### PR DESCRIPTION
**Pull request by spawn-guy**
PR for #171 
test-week.js can be later removed. as it contains only the Object-related issues. that are now incorporated into main test.js

in short: in JS new Date(); IS "Reactive" variable. new Date('some value') is a "Static" variable.
due to this - getWeekNumber - is, usually, mistaken.
it is mistaken 'cause new Date() Always holds Current time in Milliseconds since Epoch. so the value DOES change after a few "lines of code"(sic!). so the value of the Date is different in the beginning of the function and a few CPU-cycles later :|

### CODE REVIEW

File: opening_hours.js [hunk 1]
```diff
@@ -6728,7 +6728,7 @@
         function getWeekNumber(d) {
             // Copy date so don't modify original
             d = new Date(+d);
-            d.setHours(0,0,0);
+            d.setHours(0,0,0,0);
```
Comment (ypid) [inline opening_hours.js]: When I revert this change, the added unit tests still pass using NodeJS v0.10.29. Is that just because of my legacy Node version? (hm, also tested with v4.6.1 with the same result). Can you update the tests to catch that or am I missing something?

File: test-week.js [hunk 2]
```diff
@@ -0,0 +1,748 @@
+#!/usr/bin/env nodejs
```
Comment (ypid) [inline test-week.js]: You can remove the file directly. It can be left in git history. The tests are already in the main test file.

File: opening_hours.js (hunk 1 above)
Comment (spawn-guy) [inline opening_hours.js]: just tried that.
indeed, if you DO use timekeep - the bug doesn't appear.
the moment i disable timekeep - both tests fail :|

with timekeep
```
"Week range. Working with Objects not Strings. from = new Date(Date())" for "week 01-53/2 Mo-Su 07:30-08:00": PASSED
"Week range. Working with Objects not Strings. from = new Date()" for "week 01-53/2 Mo-Su 07:30-08:00": PASSED
2/2 tests passed. 0 did not pass.
```

without timekeep
```
"Week range. Working with Objects not Strings. from = new Date(Date())" for "week 01-53/2 Mo-Su 07:30-08:00": FAILED, bad duration(s): 0,0, expected 1800000,0, bad intervals: 
(none)
expected:
[ '2016.11.22 07:30', '2016.11.22 08:00', false, undefined ],
"Week range. Working with Objects not Strings. from = new Date()" for "week 01-53/2 Mo-Su 07:30-08:00": FAILED, bad duration(s): 0,0, expected 1800000,0, bad intervals: 
(none)
expected:
[ '2016.11.22 07:30', '2016.11.22 08:00', false, undefined ],
0/2 tests passed. 2 did not pass.
```

i've modified the test to reset the timekeeper and turn it on again afterwards

**Comment by spawn-guy** [comment]
also, which test do you want to be in?

this one does NOT fail with timekeeper OFF
```
test.addTest('Week range. Working with Objects not Strings. from = new Date(Date())', [
        'week ' + isOddWeekStart + '-53/2 Mo-Su 07:30-08:00',
    ], moment(new Date()).seconds(0).milliseconds(0), toTime.toDate(), [
        [toTime.hours(7).minutes(30).toDate(), toTime.hours(8).minutes(0).toDate()],
    ], 1800000, 0, false);
```

this one Does fail with timekeeper OFF
```
test.addTest('Week range. Working with Objects not Strings. from = new Date(Date())', [
        'week ' + isOddWeekStart + '-53/2 Mo-Su 07:30-08:00',
    ], moment(new Date()), toTime.toDate(), [
        [toTime.hours(7).minutes(30).toDate(), toTime.hours(8).minutes(0).toDate()],
    ], 1800000, 0, false);
```

tests against the old code

**Comment by ypid** [comment]
> also, which test do you want to be in?

Hm, both seem to pass when I add them to `test.js` with your fix. If they pass for you too, the you could add them to the `test.js` additionally?

**Comment by ypid** [comment]
Merged, thanks again!

**Comment by ypid** [comment]
Could have have a look at the test suite again? Currently, depending on the date, the three added test cases either fail all or pass all. I already tried to fix this but I was not able to make the test deterministic and at the same time make the test fail when I revert the `setHours` fix.
